src/_langs/en/updates/2015-07-22-measuring-performance-in-a-service-worker.markdown

```diff
@@ -0,0 +1,162 @@
+date: 2015-07-22
+
+title: "Measuring Performance in a Service Worker"
+description: "If you can't measure the performance of your requests in a service worker, how can you say it improves performance? Now you can answer that question with recent changes in Chrome."
+article:
+  written_on: 2015-07-22
+  updated_on: 2015-07-22
+
+authors:
+  - paulkinlan
+tags:
+  - performance
+  - serviceworker
+permalink: /updates/2015/07/measuring-performance-in-a-service-worker.html
+---
+
+Other than Jake Archibald worrying about his Developer skills rotting and 
+falling off, he made a strong case that intelligently using Service Worker you 
+can drastically improve the performance of your site or app. Watch the video for 
+an overview.
+
+[https://www.youtube.com/watch?v=d5\_6yHixpsQ](https://www.youtube.com/watch?v=d5_6yHixpsQ)
+
+If you are going to Supercharge your Page load time as Jake suggests, you really 
+need to be able to understand how Service Workers affect your page's requests.
+
+The [Resource Timing API](http://w3c.github.io/resource-timing/#widl-PerformanceResourceTiming-workerStart)
+is a critcial component in many sites RUM (Real User Monitoring)
+infrastructure because it lets you understand holistically how all of your users see
+the performance of your site ([Another use-case is detecting content injection](https://paul.kinlan.me/detecting-injected-content/)).  In short, it lets you 
+understand nearly every aspect of every web request made from your site, well, unless you have a Service Worker.
+
+Here is a quick example of how it can be used in to get a list of all requests that were made
+to a domain that is not the current domain.
+
+{% highlight Javascript %}
+var getThirdPartyRequests = function() {
+  var thirdPartyRequests = [];
+  var requests = window.performance.getEntriesByType("resource");
+  
+  var currentHost = window.location.host
+
+  for(var requestIdx = 0; requestIdx < requests.length; requestIdx++) {
+    var request = requests[requestIdx];
+    var url = new URL(request.name);
+    var host = url.host;
+
+    if(host != currentHost) {
+      thirdPartyRequests.push(request);
+    }
+  }
+  
+  return thirdPartyRequests;
+};
+{% endhighlight %}
+ 
+The Resource Timing API was created and implemented before Service Worker was a twinkle in
+an engineers eye and the above code would not be able to understand how the Service Woker
+impacted it.  This is where a recent set of changes in Chrome 45 (Beta in July 2015) will help 
+you by introducing the ability for all forms of Workers (Web and Service Worker) 
+to have access to the Resource Timing API and thus be able to let you keep on top of 
+the network performance for all your users.
+
+Note: There are futher changes coming down the pipe in later versions of Chrome so 
+this article will be kept up to date.
+
+## Accessing Performance Metrics from a Service Worker
+
+The biggest change is the addition of the `performance` object into a Workers context (Web 
```

Comment:
Note that we've added support for Resource Timing and User Timing. You can use User Timing to record own marks and measures within your worker(s).